game.cpp

```diff
@@ -46,6 +44,53 @@ bool Game::IsWall(uint8_t tileX, uint8_t tileY)
            (1 << (8 - (tileX & 0x7)));
 }
 
+void Game::CollideWithMap()
```

Comment:
Should the method name be referred to something like `DelectCollision`? Someone might collide with "wall" rather than "map."

Reply:
+ I used the word "Map" because I think "map" already includes the concept of walls and map boundaries. `CollideWithWalls` will probably work too.
+ Both `CollideWithMap` and `CollideWithWalls` should be clearer than "DetectCollision" since there might also be collisions with sprites in the future. If the collision with sprites or other objects were to be implemented, it can then be in another function like `CollideWithSprites` etc. Then, `DetectCollision` will become ambiguous because it only processes wall collisions.

Reply:
It is weird to name the method as `CollideWithWalls` since your intention was to "check" the potential collisions with the walls instead of the reaction of "colliding" something. I would stick to `CheckCollisionWithWall`.

Reply:
You're right. I will switch the name to `CheckWallCollisions`, removing the unimportant 'With'.